app/controllers/api/v2/table_preferences_controller.rb

```diff
@@ -0,0 +1,59 @@
+module Api
+  module V2
+    class TablePreferencesController < V2::BaseController
+      include Api::Version2
+      include Foreman::Controller::UserAware
+      before_action :check_user
+      before_action :find_resource, :only => [:destroy, :update, :show]
+
+      api :GET, "/users/:user_id/table_preferences", N_("List of table preferences for a user")
+      def index
+        @table_preferences = @user.table_preferences
+      end
+
+      api :GET, "/users/:user_id/table_preferences/:name", N_("List of table preferences for a table")
```

Comment:
I find the description confusing, how about "Table preferences details" ? (but not a list of table for table)